### PR TITLE
fix: update ethers deps

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Updates all ethers dependencies in response to the BN.js bug reported
for versions below 5.8.6. Also removes various extra dependencies to
reduce the total number of deps and simplify the process of updating in
the future.

### CODE REVIEW

File: contracts-bedrock/helpers/utils.ts [hunk 1]
```diff
@@ -1,19 +1,7 @@
...
-  hexConcat,
-  stripZeros,
-  zeroPad,
-  arrayify,
-  BytesLike,
-} from '@ethersproject/bytes'
-import { keccak256 } from '@ethersproject/keccak256'
-import { Zero } from '@ethersproject/constants'
-import { ContractReceipt, Event } from '@ethersproject/contracts'
+import { BigNumber, BigNumberish, BytesLike, ContractReceipt, ethers, Event } from 'ethers'
```
Comment (smartcontracts) [inline contracts-bedrock/helpers/utils.ts:1]: Using these from ethers directly reduces the number of dependencies, which makes it easier to update dependencies if we ever need to do this again in the future.

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
```
Unused devDependencies
* @ethersproject/bytes
```

File: packages/contracts/package.json [hunk 2]
```diff
@@ -98,10 +98,10 @@
     "eslint-plugin-react": "^7.24.0",
     "eslint-plugin-unicorn": "^32.0.1",
     "ethereum-waffle": "^3.3.0",
-    "ethers": "^5.5.4",
```
Comment (tynes) [inline packages/contracts/package.json:101]: `ethers` is a dev dep?

Comment (smartcontracts) [reply packages/contracts/package.json:101]: dev dev + peer dep

**Review by tynes — APPROVED** [review]

**Review by mslipper — APPROVED** [review]

**Pull request merged**